File: nom-ui/src/app/core/models/scraping-source.model.ts
export type ScrapingSourceStatus = 'Pending' | 'Approved' | 'Rejected';

export interface ScrapingSourceModel {
  id: number;
  domain: string;
  status: ScrapingSourceStatus;
  sampleUrl?: string;
  requestedByName?: string;
  createdDate: string;
  reviewedByName?: string;
  reviewedDate?: string;
  notes?: string;
}

File: nom-api/Nom.Orch/Models/Recipe/ScrapingSourceModel.cs
using System;

namespace Nom.Orch.Models.Recipe
{
    public class ScrapingSourceModel
    {
        public long Id { get; set; }
        public string Domain { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty;
        public string? SampleUrl { get; set; }
        public string? RequestedByName { get; set; }
        public DateTime CreatedDate { get; set; }
        public string? ReviewedByName { get; set; }
        public DateTime? ReviewedDate { get; set; }
        public string? Notes { get; set; }
    }

    public class ScrapingSourceReviewRequestModel
    {
        public string? Notes { get; set; }
    }
}
